server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -147,13 +147,44 @@ public class BalancedShardsAllocator implements ShardsAllocator {
         Property.NodeScope
     );
 
+    public static final Setting<Boolean> SHARDS_PER_INDEX_BALANCE = Setting.boolSetting(
```

Comment:
Where is this setting used?

Reply:
This is used to enable/disable `INDEX_SHARD_PER_NODE_BREACH_CONSTRAINT_ID` constraint . 

```
        this.weightFunction.updateAllocationConstraint(INDEX_SHARD_PER_NODE_BREACH_CONSTRAINT_ID, shardsPerIndexBalance);

```